src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -594,15 +598,21 @@ def send_raw_request(cli_ctx, method, uri, headers=None, uri_parameters=None,  #
     if not skip_authorization_header and uri.lower().startswith('https://'):
         if not resource:
             endpoints = cli_ctx.cloud.endpoints
-            from azure.cli.core.cloud import CloudEndpointNotSetException
-            for p in [x for x in dir(endpoints) if not x.startswith('_')]:
-                try:
-                    value = getattr(endpoints, p)
-                except CloudEndpointNotSetException:
-                    continue
-                if isinstance(value, six.string_types) and uri.lower().startswith(value.lower()):
-                    resource = value
-                    break
+            # If uri starts with ARM endpoint, like https://management.azure.com/,
+            # use active_directory_resource_id for resource.
+            # This follows the same behavior as azure.cli.core.commands.client_factory._get_mgmt_service_client
+            if uri.lower().startswith(endpoints.resource_manager.rstrip('/')):
+                resource = cli_ctx.cloud.endpoints.active_directory_resource_id
```

Comment:
Can it be simplified here? `cli_ctx.cloud.endpoints.active_directory_resource_id` -> `endpoints.active_directory_resource_id`

Reply:
```suggestion
            # If uri starts with ARM endpoint, like https://management.azure.com/,
            # use active_directory_resource_id for resource.
            # This follows the same behavior as azure.cli.core.commands.client_factory._get_mgmt_service_client
            if uri.lower().startswith(endpoints.resource_manager.rstrip('/')):
                resource = endpoints.active_directory_resource_id
```

Reply:
Nice catch. Changed as requested.